plugins/modules/ndo_ipsla_monitoring_policy.py

```diff
@@ -0,0 +1,359 @@
+    state = module.params.get("state")
+
+    if sla_frequency is not None and sla_frequency not in range(1, 301):
+        mso.fail_json(msg="Invalid value provided for sla_frequency: {0}; it must be in the range 1 - 300".format(sla_frequency))
+
+    if detect_multiplier is not None and detect_multiplier not in range(1, 101):
+        mso.fail_json(msg="Invalid value provided for detect_multiplier: {0}; it must be in the range 1 - 100".format(detect_multiplier))
+
+    if request_data_size is not None and request_data_size not in range(0, 17514):
+        mso.fail_json(msg="Invalid value provided for request_data_size: {0}; it must be in the range 0 - 17513".format(request_data_size))
+
+    if type_of_service is not None and type_of_service not in range(0, 256):
+        mso.fail_json(msg="Invalid value provided for type_of_service: {0}; it must be in the range 0 - 255".format(type_of_service))
+
+    if operation_timeout is not None and sla_frequency is not None and operation_timeout > (sla_frequency * 1000):
+        mso.fail_json(
+            msg="Invalid value provided for operation_timeout: {0}; must be less than or equal to: {1}".format(operation_timeout, (sla_frequency * 1000))
+        )
+
+    if threshold is not None and operation_timeout is not None and threshold > operation_timeout:
+        mso.fail_json(
+            msg="Invalid value provided for threshold: {0}; it must be less than or equal to the operation_timeout: {1}".format(threshold, operation_timeout)
+        )
+
+    if ipv6_traffic_class is not None and ipv6_traffic_class not in range(0, 256):
+        mso.fail_json(msg="Invalid value provided for ipv6_traffic_class: {0}; it must be in the range 0 - 255".format(ipv6_traffic_class))
+
+    if sla_type == "http":
+        destination_port = 80
+
+    if sla_type != "http":
+        if http_version or http_uri:
+            mso.fail_json(msg="http_version and http_uri can only be used when sla_type is 'http'")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "tenant", template)
+    mso_template.validate_template("tenantPolicy")
+    object_description = "IPSLA Monitoring Policy"
+
+    path = "/tenantPolicyTemplate/template/ipslaMonitoringPolicies"
+    existing_ipsla_policies = mso_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get("ipslaMonitoringPolicies", [])
+    if ipsla_monitoring_policy or ipsla_monitoring_policy_uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_ipsla_policies,
+            [KVPair("uuid", ipsla_monitoring_policy_uuid) if ipsla_monitoring_policy_uuid else KVPair("name", ipsla_monitoring_policy)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_ipsla_policies
+
+    if state == "present":
+        replace_data = {
```

Comment:
should we use dict() or {}? we use dict() most of the time for dict instantiation, but not sure if it matters

Reply:
samita names it mso_values, which is the variable name I followed also now, should we align on that?https://github.com/CiscoDevNet/ansible-mso/pull/574/files#diff-56139042a23bb757d511b1f82886f1ce58db3f27bc1942c0fc4831fa1c7a4223R262

Reply:
Done